lib/numbers_in_words/number_parser.rb

```diff
@@ -39,7 +39,14 @@ module NumbersInWords::NumberParser
   #3. add memory to answer,reset,  because power of ten>2    0      2000
   #4. add 1 to memory                                        1      2000
   #5. finish - add memory to answer                          0      2001
-  def parse(integers)
+  def parse(integers, only_compress = false)
+    scales_n = [100, 1000, 1000000, 1000000000, 1000000000000, 10**100] 
+    if integers.length < 2
+      return only_compress ? integers : integers.empty? ? 0 : integers[0]
+    end
+    if [] == scales_n & integers
```

Comment:
Just looked at my comment from before and I said the same thing in April. This to me is evidence enough that this is a bit confusing. Can you change it to

```ruby
  if (integers & scales_n)==[]
```

Less chance for mistaking for ([] == scales_n) && integers